lib/krb5/crypto.c

```diff
@@ -485,6 +486,8 @@ create_checksum_iov(krb5_context context,
 	ret = get_checksum_key(context, crypto, usage, ct, &dkey);
 	if (ret)
 	    return ret;
+    } else if ((flags & KRB5_CRYPTO_FLAG_ALLOW_UNKEYED_CHECKSUM) == 0) {
+	return EINVAL;
```

Comment:
Maybe KRB5KRB_AP_ERR_INAPP_CKSUM?

Reply:
For `create_checksum()`, it's a programming error. For `verify_checksum()`, it's a protocol error. Hence the different error codes :)

Reply:
Got it.